tests/e2e/lib.sh

```diff
@@ -43,42 +43,71 @@ deploy_stackrox() {
+    helm_repo_name="tmp-srox-compat"
+    helm repo add "${helm_repo_name}" https://raw.githubusercontent.com/stackrox/helm-charts/main/opensource
+    helm repo update
 
-    deploy_sensor_from_helm_charts "$target_version" ./stackrox-init-bundle.yaml
+    latest_tag=$(make tag)
 
-    echo "Sensor deployed. Waiting for sensor to be up"
-    sensor_wait
+    helm_charts=$(helm search repo ${helm_repo_name} -l)
+    central_regex="${helm_repo_name}/stackrox-central-services[ \t]*.${CENTRAL_CHART_VERSION_OVERRIDE}[ \t]*.([0-9]+\.[0-9]+\.[0-9]+)"
+    sensor_regex="${helm_repo_name}/stackrox-secured-cluster-services[ \t]*.${SENSOR_CHART_VERSION_OVERRIDE}[ \t]*.([0-9]+\.[0-9]+\.[0-9]+)"
 
-    # Bounce collectors to avoid restarts on initial module pull
-    kubectl -n stackrox delete pod -l app=collector --grace-period=0
+    # If the central version is the latest the default behavior of deploy_central() is correct for compatibility tests
+    if  [[ $helm_charts =~ $central_regex ]]; then
+        ci_export CENTRAL_CHART_DIR_OVERRIDE "./central-chart-dir-compatibility-tests"
+        central_chart="${helm_repo_name}/stackrox-central-services"
+        helm pull "${central_chart}" --version "${CENTRAL_CHART_VERSION_OVERRIDE}" --untar --untardir "${CENTRAL_CHART_DIR_OVERRIDE}"
+    elif [[ "$latest_tag" != "$CENTRAL_CHART_VERSION_OVERRIDE" ]]; then
+        echo >&2 "stackrox-central-services helm chart for version ${CENTRAL_CHART_VERSION_OVERRIDE} not found in ${helm_repo_name} repo nor is it the latest tag."
+        exit 1
+    fi
 
-    sensor_wait
+    # If the sensor version is the latest the default behavior of deploy_sensor() is incorrect, because it will deploy
+    # a sensor version to match the central version. In our tests we want to test latest sensor vs older central too,
+    # and since latest sensor is not available in the repo either the chart is created here in the elif case.
+    if [[ $helm_charts =~ $sensor_regex ]]; then
+        sensor_chart="${helm_repo_name}/stackrox-secured-cluster-services"
+        ci_export SENSOR_CHART_DIR_OVERRIDE "./sensor-chart-dir-compatibility-tests"
+        helm pull "${sensor_chart}" --version "${SENSOR_CHART_VERSION_OVERRIDE}" --untar --untardir "${SENSOR_CHART_DIR_OVERRIDE}"
+    elif [[ "$latest_tag" == "$SENSOR_CHART_VERSION_OVERRIDE" ]]; then
+        if [[ $(roxctl version) != "$latest_tag" ]]; then
+            echo >&2 "Roxctl version $(roxctl version) is not equal to latest tag ${latest_tag}"
+            exit 1
+        fi
+        ci_export SENSOR_CHART_DIR_OVERRIDE "./sensor-chart-dir-compatibility-tests"
+        roxctl helm output secured-cluster-services --image-defaults=opensource --output-dir "${SENSOR_CHART_DIR_OVERRIDE}" --remove
+        echo "Downloaded stackrox-secured-cluster-services helm chart for version ${SENSOR_CHART_VERSION_OVERRIDE} to ${SENSOR_CHART_DIR_OVERRIDE}"
+        unset SENSOR_CHART_VERSION_OVERRIDE
```

Comment:
I think it should be unnecessary.
```suggestion
```

Reply:
Yes, now that charts are downloaded locally it should be redundant